galsim/config/input_powerspectrum.py

```diff
@@ -55,6 +84,23 @@ def PowerSpectrumInit(ps, config, base):
     ps.buildGrid(grid_spacing=grid_spacing, ngrid=ngrid, center=world_center,
                  rng=base['rng'], interpolant=interpolant)
 
+    # Make sure this process gives consistent results regardless of the number of processes
+    # being used.
+    if not isinstance(ps, galsim.PowerSpectrum):
+        # Then ps is really a proxy, which means the rng was pickled, so we need to
+        # discard the same number of random calls from the one in the config dict.
```

Comment:
The multiprocessing module makes these proxy objects automatically for us.  I think it uses introspection to create something with all the same methods, but which pickles the args and kwargs to these methods and sends them through the multprocessing.Pipe back to the original object, which then calls the method.  Then the return value is again pickled and sent back through the pipe to the process that called it.

It's possible there is a more elegant way to check if we have a proxy object.  I'm certainly not a python multiprocessing expert, but I couldn't come up with anything more straightforward than this.  I suppose you could try to access an attribute and catch the exception that a proxy would raise, since it doesn't copy over all the attributes, just the public methods.  But that seems even more convoluted than this isinstance check.